docs/csharp/tutorials/nullable-reference-types.md

```diff
@@ -0,0 +1,225 @@
+---
+title: Design with nullable reference types
+description: This advanced tutorial provides an introduction to nullable reference types. You'll learn to express your design intent on when reference values may be null, and have the compiler enforce when they cannot be null.
+ms.date: 12/03/2018
+---
+# Tutorial: Express your design intent more clearly with nullable and non-nullable reference types
+
+C# 8 introduces **nullable reference types**, which complement reference types the same way nullable value types complement value types. You declare a variable to be a **nullable reference type** by appending a `?` to the type. For example, `string?` represents a nullable `string`. You can use these new types to more clearly express your design intent: some variables *must always have a value*, others *may be missing a value*.
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+> * Incorporate nullable and non-nullable reference types into your designs
+> * Enable nullable reference type checks throughout your code.
+> * Write code where the compiler enforces those design decisions.
+> * Use the nullable reference feature in your own designs
+
+## Prerequisites
+
+You’ll need to set up your machine to run .NET Core, including the C# 8 beta compiler.  The C# 8 beta compiler is available with Visual Studio 2019 preview 1, or .NET Core 3.0 preview 1.
+
+This tutorial assumes you're familiar with C# and .NET, including either Visual Studio or the .NET Core CLI.
+
+## Incorporate nullable reference types into your designs
+
+In this tutorial, you'll build a library that models running a survey. The code uses both nullable reference types and non-nullable reference types to represent the real world concepts. The survey questions would never be null. A respondent might prefer not to answer an optional question. The responses might be null in this example.
```

Comment:
survey questions would --> survey questions should, or survey questions can never be